### ISSUE TITLE
Added mputility.py to master

### ISSUE THREAD

**Issue by km-git-acc** [km-git-acc/dbn_upper_bound#25]
Have added mputility.py to the master branch. Since it is a new file, there are no conflicts with the existing code. 

Essentially, it uses a multiprecision library mpmath to compute some of the same functions used in utility.py, but also important new ones like Ht_AFE, which implements the approx functional eqn derived in Terry's blog. The need for mpmath arises while doing computations at large T heights since H_t decays exponentially with T as exp(-PI*T/8). mpmath during computation maintains a set number of significant digits as specified by us, even for very small or large numbers.

Ht_AFE is reasonably fast and can be used for large scale computations.

**Comment by sujitnair** [comment]
@km-git-acc 

This is great! Let's target to get the first million zeros of H_t for a reasonable range of t. I can help with parallelization if that helps. That way, we don't have to do the same computation twice.

**Comment by km-git-acc** [comment]
Yeah parallelization would be quite useful. Ideally, everyone can take a certain range of t in which they would operate. For eg. I can take the range between t=0.48 to 0.5, as that was what I was targeting next.

If the parallelization can work even without needing everyone to coordinate, that would be even better. People could just download a tool and start running it, and all the task allocation would be done by a central node, like how SETI@home works. But my impression is creating such a tool is a lot of work.
